instruments/abstract_instruments/instrument.py

```diff
@@ -525,6 +526,26 @@ def open_serial(cls, port=None, baud=9600, vid=None, pid=None,
         return cls(ser)
 
     @classmethod
+    def open_serial_remote(cls, url=None, baud=115200, dsrdtr=False, rtscts=False,
```

Comment:
hmm, I wonder if the better approach is to include this functionality in `open_serial` and just add more named parameters to that function

Reply:
In `open_serial`, you already offer the option to define the device either with `port` or with the `pid`/`vid`/`serial_number` combination. I think, if the logic complexity does not explode, it would be better to add another `url` parameter instead of having to maintain a separate class method just for the `for_url` approach. 
Also, even if you don't do that, one thing that smells odd to me (without having dug too deep into the api) is that the array of configuration parameters is different between `open_serial` and `open_serial_remote`, (baud/timeout/write_timeout vs baud/dsrdtr/rtscts/xonxoff/timeout).

Reply:
All excellent points